lib/jack_tokenizer.rb

```diff
@@ -74,11 +74,11 @@ def skip_whitespace
   end
 
   def skip_comments
-    if @input[@index..@index+1] == "//"
-      @index = @input.index("\n", @index) || @input.length
+    if (match = @input.match(%r{//.*$}, @index)) && match.begin(0) == @index
+      @index = match.end(0)
 
-    elsif @input[@index..@index+1] == "/*"
-      @index = @input.index("*/", @index) + 2
+    elsif (match = @input.match(%r{/\*(.|\n)*\*/}, @index)) && match.begin(0) == @index
```

Comment:
Trivia: like the `%r{…}i` and `%r{…}x` modifiers we discussed briefly, there’s also a `%r{…}m` modifier (`m` for “multiline”) that makes `.` match any character _including_ a newline, so we could’ve also written this regular expression as `%r{/\*.*\*/}m`. I didn’t suggest this during pairing because I knew I wanted you to combine the single-line-comment regex with this one, and in that case we want `.` to _not_ match a newline, so we can’t use the `m` modifier in the combined expression.